src/common/tvgArray.h

```diff
@@ -69,6 +69,11 @@ struct Array
         return true;
     }
 
+    T& operator()(size_t idx) const
```

Comment:
should we  use square brackets operator ` operator[](size_t) `?

And one more thing.
I prefer use const function to return const reference, and use non const function to return mutable reference. Just like the code in STL container.
for example:
```
const T& operator()[size_t idx] const { return data[idx]; }
T& operator()[size_t idx] { return data[idx]; }
```
Because this is meanfull and can make sure if pass the const reference array into some function, the array contents  can not be changed.

Reply:
code has been updated. hopefully you are satisfiled with that. :)